src/preamble.js

```diff
@@ -1839,7 +1849,7 @@ function addRunDependency(id) {
 #if USE_PTHREADS
   // We should never get here in pthreads (could no-op this out if called in pthreads, but that might indicate a bug in caller side,
   // so good to be very explicit)
-  assert(!ENVIRONMENT_IS_PTHREAD);
+  assert(!ENVIRONMENT_IS_PTHREAD || id === 'wasm-instantiate');
```

Comment:
This is uuuuugly.

Reply:
Did you make this change in order to get async instantiation working? It would probably be better if we avoided this altogether and follow the path `"We can just use sync instantiation in the worker."` that you wrote below?

Reply:
Actually that's not quite right; preamble.js adds the dependency regardless of whether a custom instantiator is used. We can bypass that in threads (see 9d9762) which I guess is a little bit nicer than this. I also noticed that right now the user can't override wasm-instantiate in a pthread right now (only on the main thread). Is that something we need?